src/wallet/wallet.cpp

```diff
@@ -1462,6 +1522,89 @@ bool CWallet::SetHDChain(const CHDChain &chain, bool memonly) {
     return true;
 }
 
+bool CWallet::SetMnemonicConatiner(const MnemonicConatiner& mnConatiner, bool memonly) {
+    if (!memonly && !CWalletDB(strWalletFile).WriteMnemonic(mnConatiner))
+        throw runtime_error(std::string(__func__) + ": writing chain failed");
+    mnemonicConatiner = mnConatiner;
+    return true;
+}
+
+bool CWallet::EncryptMnemonicConatiner(const CKeyingMaterial& vMasterKeyIn)
+{
+    if (!IsCrypted())
+        return false;
+
+    if (mnemonicConatiner.IsCrypted())
+        return true;
+
+    uint256 id = uint256S(hdChain.masterKeyID.GetHex());
+
+    std::vector<unsigned char> cryptedSeed;
+    if (!EncryptSecret(vMasterKeyIn, mnemonicConatiner.GetSeed(), id, cryptedSeed))
+        return false;
+    SecureVector secureCryptedSeed(cryptedSeed.begin(), cryptedSeed.end());
+    if (!mnemonicConatiner.SetSeed(secureCryptedSeed))
+        return false;
+
+    SecureString mnemonic;
+    if (mnemonicConatiner.GetMnemonic(mnemonic)) {
+        std::vector<unsigned char> cryptedMnemonic;
+        SecureVector vectorMnemonic(mnemonic.begin(), mnemonic.end());
+
+        if ((!mnemonic.empty() && !EncryptSecret(vMasterKeyIn, vectorMnemonic, id, cryptedMnemonic)))
+            return false;
+
+        SecureString secureCryptedMnemonic(cryptedMnemonic.begin(), cryptedMnemonic.end());
+        SecureString secureCryptedPassphrase;
+        if (!mnemonicConatiner.SetMnemonic(secureCryptedMnemonic, secureCryptedPassphrase, false))
+            return false;
+    }
+
+    mnemonicConatiner.SetCrypted(true);
+
+    return true;
+}
+
+bool CWallet::DecryptMnemonicConatiner(MnemonicConatiner& mnConatiner) const
```

Comment:
Typo. You need to replace Conatiner with Container all over the code

Reply:
thank you, will replace.